voltixApp/VoltixApp/Views/Keysign/KeysignDiscoveryView.swift

```diff
@@ -8,182 +8,208 @@ import SwiftUI
-            logger.info("kicked off keysign successfully")
-        }
+    .onAppear {
+      if self.appState.currentVault == nil {
+        self.currentState = .FailToStart
+        return
+      }
+      if let localPartyID = appState.currentVault?.localPartyID, !localPartyID.isEmpty {
+        self.localPartyID = localPartyID
+      } else {
+        self.localPartyID = Utils.getLocalDeviceIdentity()
+      }
+    }
+    .task {
+      // start the mediator , so other devices can discover us
+      self.mediator.start()
+      self.startKeysignSession()
+      Task {
+        repeat {
+          self.getParticipants()
+          try await Task.sleep(nanoseconds: 1_000_000_000)  // wait for a second to continue
+        } while self.discoverying
+      }
     }
+    .onDisappear {
+      logger.info("mediator server stopped")
+      self.discoverying = false
+      self.mediator.stop()
+    }
+  }
 
-    private func startKeysignSession() {
-        let urlString = "\(self.serverAddr)/\(self.sessionID)"
-        let body = [self.localPartyID]
-        Utils.sendRequest(urlString: urlString, method: "POST", body: body) { success in
-            if success {
-                logger.info("Started session successfully.")
-            } else {
-                logger.info("Failed to start session.")
+  private func startKeysign(allParticipants: [String]) {
+    let urlString = "\(self.serverAddr)/start/\(self.sessionID)"
+    Utils.sendRequest(urlString: urlString, method: "POST", body: allParticipants) { _ in
+      logger.info("kicked off keysign successfully")
+    }
+  }
+
+  private func startKeysignSession() {
+    let urlString = "\(self.serverAddr)/\(self.sessionID)"
+    let body = [self.localPartyID]
+    Utils.sendRequest(urlString: urlString, method: "POST", body: body) { success in
+      if success {
+        logger.info("Started session successfully.")
+      } else {
+        logger.info("Failed to start session.")
+      }
+    }
+  }
+
+  private func getParticipants() {
+    let urlString = "\(self.serverAddr)/\(self.sessionID)"
+    Utils.getRequest(
+      urlString: urlString, headers: [String: String](),
+      completion: { result in
+        switch result {
+        case .success(let data):
+          if data.isEmpty {
+            logger.error("No participants available yet")
+            return
+          }
+          do {
+            let decoder = JSONDecoder()
+            let peers = try decoder.decode([String].self, from: data)
+
+            for peer in peers {
+              if !self.peersFound.contains(peer) {
+                self.peersFound.append(peer)
+              }
             }
+          } catch {
+            logger.error("Failed to decode response to JSON: \(error)")
+          }
+        case .failure(let error):
+          logger.error("Failed to start session, error: \(error)")
+          return
         }
+      })
+  }
+
+  func getQrImage(size: CGFloat) -> CGImage? {
```

Comment:
the one on main branch is better , it returns Image, which is support on both ios and macos

Reply:
I've changed it because it was not working for me while compiling. If you feel the current one is better then it is fine.